python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -5,209 +5,205 @@
 
 from __future__ import annotations
 
-import copy
+import dataclasses
+import enum
 import json
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Literal
 
 if TYPE_CHECKING:
-    from collections.abc import MutableMapping, Sequence
-
     from typing_extensions import Self
 
+    import polars as pl
+
 __all__ = ["ConfigOptions"]
 
 
-class ConfigOptions:
-    """
-    GPUEngine configuration-option manager.
+@dataclasses.dataclass(frozen=True)
+class ParquetOptions:
+    """Configuration for the cudf-polars Parquet engine."""
 
-    This is a convenience class to help manage the nested
-    dictionary of user-accessible `GPUEngine` options.
-    """
+    chunked: bool = True
+    chunk_read_limit: int = 0
+    pass_read_limit: int = 0
 
-    __slots__ = ("_hash_value", "config_options")
-    _hash_value: int
-    config_options: dict[str, Any]
-    """The underlying (nested) config-option dictionary."""
+    def __post_init__(self) -> None:
```

Comment:
I've added some basic validation to these dataclasses, since they are coming from arbitrary user input.

That's also why I've used `Enums` for some things that I'd normally just use `Literal` for: by putting them through an enum we'll validate that the user-provided value is in the set of expected values.